react/modules/pickInstallments.ts

```diff
@@ -92,3 +95,108 @@ function pickMaxOption(
 
   return clusteredInstallments[biggestInstallmentsOptionKey]
 }
+
+function applyFiltersToInstallmentsList(
+  installmentsList: ProductTypes.Installment[],
+  filteringRules: {
+    paymentSystemName?: string
+    installmentsQuantity?: number
+  }
+) {
+  let filteredInstallmentsList = installmentsList
+
+  if (filteringRules.paymentSystemName) {
+    filteredInstallmentsList = filteredInstallmentsList.filter(
+      installmentsOption =>
+        installmentsOption.PaymentSystemName ===
+        filteringRules.paymentSystemName
+    )
+  }
+
+  if (filteringRules.installmentsQuantity) {
+    filteredInstallmentsList = filteredInstallmentsList.filter(
+      installmentsOption =>
+        installmentsOption.NumberOfInstallments ===
+        filteringRules.installmentsQuantity
+    )
+  }
+
+  return filteredInstallmentsList
+}
+
+export function pickMaxInstallmentsOption(
+  installmentsList: ProductTypes.Installment[],
+  filteringRules?: {
+    paymentSystemName?: string
+    installmentsQuantity?: number
+  }
+) {
+  let filteredInstallmentsList = installmentsList
+
+  if (filteringRules) {
+    filteredInstallmentsList = applyFiltersToInstallmentsList(
+      installmentsList,
+      filteringRules
+    )
+  }
+
+  let [maxInstallmentOption] = filteredInstallmentsList
+
+  filteredInstallmentsList.forEach(installmentOption => {
+    if (
+      installmentOption.NumberOfInstallments >
+      maxInstallmentOption.NumberOfInstallments
+    ) {
+      maxInstallmentOption = installmentOption
+    }
+  })
+
+  return maxInstallmentOption
+}
+
+export function pickMaxInstallmentsOptionWithNoInterest(
```

Comment:
I think this algorithm can be more simple. I will try to scratch it:
So you want to pick the highest installment that doesn't have interest, right?

You can follow these steps:
1. filter the installmentsList to remove all installments that have interest
2. : 
I. If the result of the last step (1) is an empty list, then return `pickMaxInstallmentsOption` passing the original `installmentsList`
II. otherwise, return `pickMaxInstallmentOption` passing the filtered list of the last step (1)

By doing this you avoid the need to find the first no interest option, doing that logic, and enhance the readability.